crates/goose/src/systems/non_developer.rs

```diff
@@ -0,0 +1,971 @@
+            .save_to_cache(result.as_bytes(), operation, extension)
+            .await?;
+
+        // Register as a resource
+        let uri = Url::from_file_path(&cache_path)
+            .map_err(|_| AgentError::ExecutionError("Invalid cache path".into()))?
+            .to_string();
+
+        let resource = Resource::new(
+            uri.clone(),
+            Some("text".to_string()),
+            Some(cache_path.to_string_lossy().into_owned()),
+        )
+        .map_err(|e| AgentError::ExecutionError(e.to_string()))?;
+
+        self.active_resources.lock().unwrap().insert(uri, resource);
+
+        Ok(vec![Content::text(format!(
+            "Result saved to: {}",
+            cache_path.display()
+        ))])
+    }
+
+    // Implement quick_script tool functionality
+    async fn quick_script(&self, params: Value) -> AgentResult<Vec<Content>> {
+        let language = params
+            .get("language")
+            .and_then(|v| v.as_str())
+            .ok_or_else(|| AgentError::InvalidParameters("Missing 'language' parameter".into()))?;
+
+        let script = params
+            .get("script")
+            .and_then(|v| v.as_str())
+            .ok_or_else(|| AgentError::InvalidParameters("Missing 'script' parameter".into()))?;
+
+        let save_output = params
+            .get("save_output")
+            .and_then(|v| v.as_bool())
+            .unwrap_or(false);
+
+        // Create a temporary directory for the script
+        let script_dir = tempfile::tempdir().map_err(|e| {
+            AgentError::ExecutionError(format!("Failed to create temporary directory: {}", e))
+        })?;
+
+        let command = match language {
+            "shell" => {
+                let script_path = script_dir.path().join("script.sh");
+                fs::write(&script_path, script).map_err(|e| {
+                    AgentError::ExecutionError(format!("Failed to write script: {}", e))
+                })?;
+
+                fs::set_permissions(&script_path, fs::Permissions::from_mode(0o755)).map_err(
+                    |e| {
+                        AgentError::ExecutionError(format!(
+                            "Failed to set script permissions: {}",
+                            e
+                        ))
+                    },
+                )?;
+
+                script_path.display().to_string()
+            }
+            "applescript" => {
+                if std::env::consts::OS != "macos" {
+                    return Err(AgentError::ExecutionError(
+                        "AppleScript is only supported on macOS".into(),
+                    ));
+                }
+
+                let script_path = script_dir.path().join("script.scpt");
+                fs::write(&script_path, script).map_err(|e| {
+                    AgentError::ExecutionError(format!("Failed to write script: {}", e))
+                })?;
+
+                format!("osascript {}", script_path.display())
+            }
+            "ruby" => {
```

Comment:
LLMs tend to be better at writing python + i think it comes with default Mac, so curious why `ruby` is used here.

Reply:
I think that was true, less so now with newer gen LLMs, also on macos python is not default (I know as I just bought a new one!) but, strangely, ruby is! (linux is other way around). So really just as ruby is standard on macos with 0 dev tools installed, out of the box

Reply:
makes sense